ada/language/ast.py

```diff
@@ -356,7 +356,18 @@ def referenced_decl_internal_helper(ref_var=T.LogicVar,
             )),
             Self.logic_val(Entity, ref_var, try_immediate)
                 .value.cast_or_raise(T.BasicDecl.entity)
-        )
+        ).then(lambda x: x.match(
+            # If the logic variable is bound to a GenericSubpInternal, retrieve
+            # the instantiation leading to it instead.
+            lambda g=T.GenericSubpInternal: entity_no_md(
+                type=T.BasicDecl,
+                node=g.info.rebindings.new_env.env_node
+                .cast_or_raise(T.GenericInstantiation),
+                rebindings=x.info.rebindings.get_parent,
```

Comment:
A comment here would be helpful, explaining you're removing the instantiation from the list of rebindings.

Reply:
ok